File: spec/php/DefaultEndianModTest.php
<?php
namespace Kaitai\Struct\Tests;

class DefaultEndianModTest extends TestCase {
    public function testDefaultEndianMod() {
        $r = DefaultEndianMod::fromFile(self::SRC_DIR_PATH . "/fixed_struct.bin");

        $this->assertEquals(0x4b434150, $r->main->one);
        $this->assertEquals(-52947, $r->main->nest->two);
        $this->assertEquals(0x5041434b, $r->main->nestBe->two);
    }
}

File: spec/go/default_endian_mod_test.go
package spec

import (
	"os"
	"testing"

	"github.com/kaitai-io/kaitai_struct_go_runtime/kaitai"
	"github.com/stretchr/testify/assert"

	. "test_formats"
)

func TestDefaultEndianMod(t *testing.T) {
	f, err := os.Open("../../src/fixed_struct.bin")
	if err != nil {
		t.Fatal(err)
	}
	s := kaitai.NewStream(f)

	var r DefaultEndianMod
	err = r.Read(s, &r, &r)
	if err != nil {
		t.Fatal(err)
	}

	assert.EqualValues(t, r.Main.One, 0x4b434150)
	assert.EqualValues(t, r.Main.Nest.Two, -52947)
	assert.EqualValues(t, r.Main.NestBe.Two, 0x5041434b)
}
